resource-manager/src/main/java/org/snomed/module/storage/RF2Service.java

```diff
@@ -0,0 +1,95 @@
+package org.snomed.module.storage;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+public class RF2Service {
+    public static final Integer MODULE_ID = 3;
+    public static final Integer REFERENCED_COMPONENT_ID = 5;
+    public static final Integer TARGET_EFFECTIVE_TIME = 7;
+
+    public Set<String> getUniqueModuleIds(File file) {
+        if (file == null) {
+            return Collections.emptySet();
+        }
+
+        List<List<String>> rows = getRows(file, null, MODULE_ID);
+        Set<String> result = new HashSet<>();
+        for (List<String> row : rows) {
+            result.addAll(row);
+        }
+
+        return result;
+    }
+
+    public Set<RF2Row> getMDRS(File file) {
+        if (file == null) {
+            return Collections.emptySet();
+        }
+
+        List<List<String>> rows = getRows(file, "der2_ssRefset_ModuleDependency", REFERENCED_COMPONENT_ID, TARGET_EFFECTIVE_TIME);
+        Set<RF2Row> rf2Rows = new HashSet<>();
+        for (List<String> row : rows) {
+            rf2Rows.add(
+                    new RF2Row()
+                            .addRow(REFERENCED_COMPONENT_ID, row.get(0))
+                            .addRow(TARGET_EFFECTIVE_TIME, row.get(1))
+            );
+        }
+
+        return rf2Rows;
+    }
+
+    private List<List<String>> getRows(File file, String fileNamePattern, Integer... columnIndices) {
```

Comment:
@pgwilliams Generic method that returns any number of columns from either all files or a specific file. This is used to find the MDRS entries and the unique module entries.